### PR TITLE
MSC2197 Search Filters over Federation

### PR DESCRIPTION
**Pull request by reivilibre**
No rush on this one, but it is a necessary component of the upcoming efficient room directory searches.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

### CODE REVIEW

**Comment by richvdh** [comment]
for links: https://github.com/matrix-org/matrix-doc/pull/2197

File: changelog.d/5859.feature [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Add unstable support for MSC2197 (filtered search requests over federation), in
+order to allow upcoming room directory query performance improvements.
```
Comment (erikjohnston) [inline changelog.d/5859.feature]: I think we want this on one line? I don't know if towncrier's formatter will handle multiline stuff sanely?

Comment (richvdh) [reply changelog.d/5859.feature]: yes please. Towncrier manages it ok, but github treats the newline as a hard break when it is C&Ped into the release notes, which doesn't look pretty.

**Comment by erikjohnston** [comment]
This is fine, but the FCP for the MSC is almost up so I vote we skip implementing `/unstable` and stick it on the stable prefix from the get go.

**Comment by reivilibre** [comment]
> This is fine, but the FCP for the MSC is almost up so I vote we skip implementing `/unstable` and stick it on the stable prefix from the get go.

Yup, I agree. I wasn't expecting to get FCP so quickly.

I'll modify this branch soon with that change, and the Towncrier newline change.

**Pull request merged**